openlibrary/components/BarcodeScanner.vue

```diff
@@ -0,0 +1,374 @@
+            if (isbn === this.lastISBN) return;
+            if (this.seenISBN.has(isbn)) return;
+
+            if (this.returnTo) {
+                location = this.returnTo.replace('$$$', isbn);
+            }
+            this.isbnList.unshift({isbn: isbn, cover: tentativeCoverUrl});
+            this.seenISBN.add(isbn);
+        },
+
+        isBarcodeISBN(code) {
+            return code.startsWith('97');
+        },
+
+    },
+    async mounted() {
+        await this.start();
+        this.canvasInactive = false;
+    }
+}
+</script>
+
+<style lang="less">
+  @keyframes camera-flash {
+  0% { filter: brightness(1.3); }
+  100% { filter: brightness(1.2); }
+  }
+
+  @keyframes slideUp {
+    0% { transform: translateY(50%); opacity: .5; }
+    100% { transform: translateY(0); opacity: 1; }
+  }
+
+  @keyframes shiftRight {
+    0% { transform: translateX(-100%); }
+    100% { transform: translateX(0); }
+  }
+
+  #page-barcodescanner {
+    display: flex;
+    flex-direction: column;
+    // stylelint-disable declaration-block-no-duplicate-properties
+    // Fallback browsers that don't support dvh
+    height: calc(100vh - 60px);
+    height: 100dvh; // stylelint-disable-line unit-no-unknown
+    // stylelint-enable declaration-block-no-duplicate-properties
+  }
+
+  .viewport {
+    position: relative;
+    flex: 1;
+    min-height: 0;
+    video, canvas {
+      width: 100%;
+      height: 100%;
+      object-fit: cover;
+    }
+    canvas {
+      position: absolute;
+      top: 0;
+      left: 0;
+    }
+
+    video[controls] + canvas {
+      pointer-events: none;
+    }
+  }
+
+  .barcodescanner__toolbar {
+    font-family: "Lucida Grande", Veranda, Geneva, Helvetica, Arial, sans-serif;
+    position: relative;
+    background: hsl(0, 0%, 100%);
+  }
+
+  .glass-button {
+    cursor: pointer;
+
+    border-radius: 10px;
+    // stylelint-disable-next-line sh-waqar/declaration-use-variable
+    color: rgba(255,255,255, .8);
+    background: rgba(0,0,0,.5);
+    border: 1px solid rgba(255,255,255, .45);
+    font-size: .8em;
+    backdrop-filter: blur(8px);
+
+    box-shadow: 0 0 3px 0 rgba(0,0,0,.4);
+    &:disabled {
+      opacity: .5;
+    }
+  }
+
+  .icon-button {
+    display: flex;
+    align-items: center;
+    padding-right: 10px;
+  }
+
+  details.barcodescanner__advanced {
+    position: absolute;
+    bottom: 100%;
+    right: 0;
+
+    // For slide up effect
+    transform: translateY(100%);
+    transition: transform .2s;
+    > :not(summary) {
+      transition: opacity .2s;
+      opacity: 0;
+    }
+    &[open] {
+      transform: translateY(0);
+    }
+    &[open] > :not(summary) {
+      opacity: 1;
+    }
+
+    & > summary {
+      position: absolute;
+      bottom: 100%;
+      right: 0;
+      margin: 5px;
+
+      &::marker {
+        content: none;
+      }
+
+      &::-webkit-details-marker {
+        display: none;
+      }
+
+      svg {
+        width: 32px;
+        height: 32px;
+        padding: 6px;
```

Comment:
Fix settings cog too large in barcode scanner

```suggestion
        padding: 6px;
        box-sizing: border-box;
```